Frameworks/Foundation/NSOperation.mm

```diff
@@ -77,41 +61,88 @@ + (BOOL)automaticallyNotifiesObserversForKey:(NSString*)key {
     return NO;
 }
 
-/**
- @Status Interoperable
-*/
-+ (id)allocWithZone:(NSZone*)zone {
-    NSOperation* ret = [super allocWithZone:zone];
+- (void)_checkReady {
+    bool newReady = YES;
 
-    ret->priv = new NSOperationPriv();
+    // If cancelled, skip this logic and set _ready to YES. Otherwise check dependencies.
+    if (![self isCancelled]) {
+        _dependenciesLock.lock();
+        int count = [_dependencies count];
```

Comment:
why?

---
In reply to: [69624996](https://github.com/Microsoft/WinObjC/pull/602#discussion_r69624996) [](ancestors = 69624996,69623694)